src/test/java/io/confluent/connect/jdbc/sink/DbStructureTest.java

```diff
@@ -41,6 +61,97 @@ public void testMissingFieldsWithDifferentCase() {
     assertTrue(missingFields(sinkRecords("AaA", "bBb"), columns("aAa", "BbB")).isEmpty());
   }
 
+  @Test (expected = SchemaMismatchException.class)
+  public void testMissingTableNoAutoCreate() throws Exception {
+    structure.create(mock(JdbcSinkConfig.class), mock(Connection.class), mock(TableId.class),
+        fieldsMetadata);
+  }
+
+  @Test (expected = SchemaMismatchException.class)
+  public void testAlterNoAutoEvolve() throws Exception {
+    TableDefinition tableDefinition = mock(TableDefinition.class);
+    when(tableDefinitions.get(any(), any())).thenReturn(tableDefinition);
+    when(tableDefinition.type()).thenReturn(TableType.TABLE);
+
+    SinkRecordField sinkRecordField = new SinkRecordField(
+        Schema.OPTIONAL_INT32_SCHEMA,
+        "test",
+        false
+    );
+    Set<SinkRecordField> missingFields = new HashSet<>();
+    missingFields.add(sinkRecordField);
+
+    doReturn(missingFields).when(spyStructure).missingFields(any(), any());
+
+    spyStructure.amendIfNecessary(mock(JdbcSinkConfig.class), mock(Connection.class), mock(TableId.class),
+        fieldsMetadata, 5);
+  }
```

Comment:
I don't think we need to spy on the class we're testing (which makes me a little nervous). We can do something like this instead:

```java
  @Test (expected = SchemaMismatchException.class)
  public void testAlterNoAutoEvolve() throws Exception {
    TableDefinition tableDefinition = mock(TableDefinition.class);
    when(tableDefinitions.get(any(), any())).thenReturn(tableDefinition);
    when(tableDefinition.type()).thenReturn(TableType.TABLE);
    when(tableDefinition.columnNames()).thenReturn(Collections.emptySet());

    SinkRecordField sinkRecordField = new SinkRecordField(
        Schema.OPTIONAL_INT32_SCHEMA,
        "test",
        false
    );
    fieldsMetadata = new FieldsMetadata(
        Collections.emptySet(),
        Collections.singleton(sinkRecordField.name()),
        Collections.singletonMap(sinkRecordField.name(), sinkRecordField));

    structure.amendIfNecessary(mock(JdbcSinkConfig.class), mock(Connection.class), mock(TableId.class),
        fieldsMetadata, 5);
  }
```

where we create an actual `FieldsMetadata` object that we pass to `DbStructure::amendIfNecessary` instead of mocking the `missingFields` method.

Hopefully this same approach can be taken with the other tests here and we can get rid of the `spyStructure` field; let me know if that doesn't work.